spec/registrant_spec.rb

```diff
@@ -0,0 +1,103 @@
+            expect(@facility_1.administer_written_test(@registrant_1)).to eq(false)
+            expect(@registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+            expect(@facility_1.add_service('Written Test')).to eq(["Written Test"])
+            expect(@facility_1.administer_written_test(@registrant_1)).to eq(true)
+            expect(@registrant_1.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+
+            expect(@registrant_2.age).to eq(16)
+            expect(@registrant_2.permit?).to eq(false)
+            expect(@facility_1.administer_written_test(@registrant_2)).to eq(false)
+            @registrant_2.earn_permit
+            expect(@facility_1.administer_written_test(@registrant_2)).to eq(true) 
+            expect(@registrant_2.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+
+            expect(@registrant_3.age).to eq(15)
+            expect(@registrant_3.permit?).to eq(false)
+            expect(@facility_1.administer_written_test(@registrant_3)).to eq(false)
+            @registrant_3.earn_permit
+            expect(@facility_1.administer_written_test(@registrant_3)).to eq(false)
+            expect(@registrant_3.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+
+            expect(@facility_1.administer_road_test(@registrant_3)).to eq(false)
+            @registrant_3.earn_permit
+            expect(@facility_1.administer_road_test(@registrant_3)).to eq(false)
+            expect(@registrant_3.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+
+            expect(@facility_1.administer_road_test(@registrant_1)).to eq(false)
+            expect(@facility_1.add_service('Road Test')).to eq(["Written Test", "Road Test"])
+            expect(@facility_1.administer_road_test(@registrant_1)).to eq(true)
+            expect(@registrant_1.license_data).to eq({:written=>true, :license=>true, :renewed=>false})
+            
+            expect(@facility_1.administer_road_test(@registrant_2)).to eq(true)
+            expect(@registrant_2.license_data).to eq({:written=>true, :license=>true, :renewed=>false})
+
+            expect(@facility_1.renew_drivers_license(@registrant_1)).to eq(false)
+            expect(@facility_1.add_service('Renew License')).to eq(["Written Test", "Road Test", "Renew License"])
+            expect(@facility_1.renew_drivers_license(@registrant_1)).to eq(true)
+            expect(@registrant_1.license_data).to eq({:written=>true, :license=>true, :renewed=>true})
+
+            expect(@facility_1.renew_drivers_license(@registrant_3)).to eq(false)
+            expect(@registrant_3.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+            expect(@facility_1.renew_drivers_license(@registrant_2)).to eq(true)
+            expect(@registrant_2.license_data).to eq({:written=>true, :license=>true, :renewed=>true})
+        end
+    end
+end 
```

Comment:
just a general comment on testing -- your tests are looking really good. I like that you have a lot of different instance that you test your methods on, that is what makes robust tests!